tool_path_planner/src/utilities.cpp

```diff
@@ -343,33 +343,74 @@ ToolPaths splitSegments(ToolPaths tool_paths, double max_segment_length)
   return new_tool_paths;
 }
 
-ToolPaths addExtraPaths(ToolPaths tool_paths, double offset_distance)
+ToolPaths reverseOddRasters(ToolPaths& tool_paths, RasterStyle raster_style)
+{
+  ToolPaths new_tool_paths;
+  int is_odd = 0;
+  int q = 0;
+  for (auto tool_path : tool_paths)
+  {
+    ToolPath new_tool_path;
+    if (!is_odd)
+    {
+      for (auto seg : tool_path)
+      {
+        new_tool_path.push_back(seg);
+      }
+    }  // end if !is_odd
+    else
+    {  // reverse order of segments,
+      for (long int i = tool_path.size() - 1; i >= 0; i--)
+      {
+        ToolPathSegment new_segment;
+        for (long int j = tool_path[i].size() - 1; j >= 0; j--)
+        {
+          // rotate around z-axis of waypoint by 180 degrees (PI)
+          Eigen::Isometry3d waypoint = tool_path[i][j];
+          if (raster_style == FLIP_ORIENTATION_ON_REVERSE_STROKES)
+            waypoint.rotate(Eigen::AngleAxisd(4 * atan(1), Eigen::Vector3d::UnitZ()));
+          new_segment.push_back(waypoint);
+        }
+        new_tool_path.push_back(new_segment);
+      }
+    }
+    new_tool_paths.push_back(new_tool_path);
+    is_odd = (is_odd + 1) % 2;
```

Comment:
`is_odd` was declared as an `int`, but is being used as a `bool`.  Perhaps it would be simpler to declare it as a bool?  That way, flipping it would be much simpler at the end:
```
bool is_odd = 0;
...
if (!is_odd)
...
is_odd = !is_odd;
```